ui-tests/tests/widget.test.ts

```diff
@@ -0,0 +1,164 @@
+    await page.locator('#deleteConnBtn_default').click();
+    await page.locator('#deleteConnectionButton').click();
+
+    expect(page.locator('#connectionsButtonsContainer')).toBeEmpty();
+
+});
+
+
+test('test edit connection', async ({ page }) => {
+    await createDefaultConnection(page);
+
+    // click on edit connection button, edit, and confirm
+    await page.locator('#editConnBtn_default').click();
+    await page.locator('#database').fill('duck.db');
+    await page.locator('#updateConnectionFormButton').click();
+
+
+    // check that connection is still there
+    let connectionButton = page.locator('#connBtn_default')
+    await connectionButton.waitFor();
+    await expect(connectionButton).toContainText('Connected');
+    await expect(page.locator('.connection-name')).toContainText('default');
+
+    await page.notebook.addCell("code", "%%sh\ncat connections.ini")
+    await page.notebook.runCell(1);
+
+    let output
+    output = await page.notebook.getCellTextOutput(1);
+    await expect(output[0]).toContain('database = duck.db');
+});
+
+
+test('test edit connection alias', async ({ page }) => {
+    await createDefaultConnection(page);
+
+    // click on edit connection button, edit, and confirm
+    await page.locator('#editConnBtn_default').click();
+    await page.locator('#connectionName').fill('duckdb');
+    await page.locator('#updateConnectionFormButton').click();
+
+
+    // check that there is only one connection
+    let connectionsDiv = page.locator('#connectionsButtonsContainer')
+    await connectionsDiv.waitFor();
+
+    const childDivs = connectionsDiv.locator('> div');
+    const innerDivCount = await childDivs.count();
+    expect(innerDivCount).toBe(1);
+});
+
+
+test('test error if creates connection with existing name', async ({ page }) => {
+    await createDefaultConnection(page);
+
+    await page.locator('#createNewConnection').click();
+    await page.locator('#connectionName').fill('default');
+    await page.locator('#createConnectionFormButton').click();
+
+    await expect(page.locator('.user-error-message')).toContainText('A connection named default already exists in your connections file');
+});
+
+
+test('test error if edit connection with existing name', async ({ page }) => {
+    // create default connection
+    await createDefaultConnection(page);
+
+    // create a new connection
+    await page.locator('#createNewConnection').click();
+    await page.locator('#connectionName').fill('duckdb');
+    await page.locator('#createConnectionFormButton').click();
+
+    // try to rename it to default, this should fail
+    await page.locator('#editConnBtn_duckdb').click();
+    await page.locator('#connectionName').fill('default');
+    await page.locator('#updateConnectionFormButton').click();
+
+
+    await expect(page.locator('.user-error-message')).toContainText('A connection named default already exists in your connections file');
+});
```

Comment:
I'm not sure if it is possible but we can add 2 more tests

1. test error if editing existing connection with a failing connection
2. test error if creating a new connection with a failing connection

However, I tested it manually and it works.

Reply:
added both tests https://github.com/ploomber/jupysql-plugin/pull/70/commits/035e80b0328e81305989242b38fc9cb7b371aff4